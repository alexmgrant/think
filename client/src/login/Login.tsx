import React, { FormEvent, useState } from 'react';

import Card from '../components/card/Card';
import Button from '../components/button/Button';
import { getEventValue } from '../common/Utils';
import { authLogin, authGithub } from '../common/ApiUtils';

const handleGithubAuth = () => {
  authGithub();
};

const emptyString = (string: string): boolean => string === '';

const Login = (props: { setJwt: Function }) => {
  const { setJwt } = props;
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const inputsValid = !(!emptyString(email) && !emptyString(password));

  const handleLogin = async (event: FormEvent) => {
    event.preventDefault();
    const response = await authLogin({ email, password });
    const token = response.data;

    setJwt(token);
    localStorage.setItem('token', token);
  };

  return (
    <section>
      <Card>
        <h4>ðŸ‘‹ Hey! you need to login.</h4>
        <form onSubmit={handleLogin}>
          <label>
            <p>Email</p>
            <input
              value={email}
              onInput={(event) => setEmail(getEventValue(event))}
              type="text"
            />
          </label>
          <label>
            <p>Password</p>
            <input
              value={password}
              onInput={(event) => setPassword(getEventValue(event))}
              type="password"
            />
          </label>
          <Button type="submit" disabled={inputsValid}>
            Login
          </Button>
        </form>
        <p>Or...</p>
        <Button onClick={handleGithubAuth}>
          Login with Github
          <img
            src={`${process.env.PUBLIC_URL}/octocat.png`}
            alt="Github Octocat"
          />
        </Button>
      </Card>
    </section>
  );
};
export default Login;
