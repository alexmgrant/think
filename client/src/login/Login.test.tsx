import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Login from './Login';

let container: Element | null = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container as Element);
  (container as Element).remove();
  container = null;
});

it('should render Login', () => {
  act(() => render(<Login setJwt={() => {}} />, container));

  expect(pretty((container as Element).innerHTML)).toMatchSnapshot();
});
