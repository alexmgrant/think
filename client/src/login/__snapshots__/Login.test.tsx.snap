// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login 1`] = `
"<section>
  <section class=\\"Card\\">
    <h4>ðŸ‘‹ Hey! you need to login.</h4>
    <form><label>
        <p>Email</p><input type=\\"text\\" value=\\"\\">
      </label><label>
        <p>Password</p><input type=\\"password\\" value=\\"\\">
      </label><button class=\\"Button\\" type=\\"submit\\" disabled=\\"\\">Login</button></form>
    <p>Or...</p><button class=\\"Button\\" type=\\"button\\">Login with Github<img src=\\"/octocat.png\\" style=\\"width: 25px;\\" alt=\\"Github Octocat\\"></button>
  </section>
</section>"
`;
